relayer/relays/parachain/beefy-listener.go

```diff
@@ -634,45 +628,51 @@ func (li *BeefyListener) scanForCommitments(
 					return nil, fmt.Errorf("basic channel commitment hash in digest item does not match the one in the Committed event")
 				}
 
-				for _, accountAndNonce := range basicAccountsAndNoncesToFind {
-					_, shouldCheckAccount := basicAccountsToCheck[accountAndNonce.account]
+				for bundleIndex, bundle := range events.Basic.Bundles {
+					_, shouldCheckAccount := scanBasicChannelAccounts[bundle.Account]
 					if !shouldCheckAccount {
 						continue
 					}
-
-					// Only consider message bundles for the accounts we're interested in
-					bundleIndex := bundleIndexForAccount(events.Basic.Bundles, &accountAndNonce.account)
-					if bundleIndex == -1 {
-						continue
-					}
-					bundle := events.Basic.Bundles[bundleIndex]
-
-					basicChannelBundleProof, err = fetchBundleProof(li.parachainConnection.API(), digestItem, bundleIndex)
-					if err != nil {
-						return nil, err
-					}
+					nonceToFind := basicAccountsAndNoncesToFind[bundle.Account]
 
 					bundleNonceBigInt := big.Int(bundle.Nonce)
 					bundleNonce := bundleNonceBigInt.Uint64()
 
 					// This case will be hit if basicNonceToFind has not
 					// been committed yet. Channels emit commitments every N
 					// blocks.
-					if bundleNonce < accountAndNonce.nonceToFind {
+					if bundleNonce < nonceToFind {
 						log.Debugf(
 							"Halting scan for account '%v'. Messages not committed yet on basic channel",
-							types.HexEncodeToString(accountAndNonce.account[:]),
+							types.HexEncodeToString(bundle.Account[:]),
 						)
-						delete(basicAccountsToCheck, accountAndNonce.account)
-						scanBasicChannelDone = len(basicAccountsToCheck) == 0
+						delete(scanBasicChannelAccounts, bundle.Account)
```

Comment:
Not directly related to this bit of code, but this method has become deeply nested and a little unwieldy (no doubt from all the new functionality we've introduced! 😄 ). It could benefit from splitting chunks of logic like these into smaller methods to make it more digestible.

Reply:
Alistair mentioned this too, I'll find a way to make this clearer 👌